src/otx/algo/common/utils/bbox_overlaps.py

```diff
@@ -142,15 +144,24 @@ def bbox_overlaps(
         >>> assert tuple(bbox_overlaps(nonempty, empty).shape) == (1, 0)
         >>> assert tuple(bbox_overlaps(empty, empty).shape) == (0, 0)
     """
+    if not (bboxes1.size(-1) == 4 or bboxes1.size(0) == 0) or not (bboxes2.size(-1) == 4 or bboxes2.size(0) == 0):
+        msg = "The last dimension of bboxes must be 4."
+        raise ValueError(msg)
+
+    if bboxes1.shape[:-2] != bboxes2.shape[:-2]:
```

Comment:
I think `bboxes1.size(0)` is more noticeable than `bboxes1.shape[:-2]` when checking batch dimension, what do you think?

Reply:
This function supports both batched (B, N, 4) and non-batched (N, 4) operation.  When a tensor doesn't not have batch dim it retunrs an empty tensor. This can not be done with tensor.size(0)